### PR TITLE
Users can have array options

### PR DESCRIPTION
**Pull request by jpakkane**
This is one of those things that I though had been exposed ages ago but apparently weren't. They were only used for Windows linker flags. Expose them now properly with tests.

### CODE REVIEW

File: docs/markdown/Build-options.md [hunk 1]
```diff
@@ -16,18 +16,20 @@ Here is a simple option file.
 option('someoption', type : 'string', value : 'optval', description : 'An option')
 option('other_one', type : 'boolean', value : false)
 option('combo_opt', type : 'combo', choices : ['one', 'two', 'three'], value : 'three')
+option('array_opt', type : 'stringarray', value : ['one', 'two'])
```
Comment (1ace) [inline docs/markdown/Build-options.md]: `s/value/choices/`, right?

Comment (1ace) [reply docs/markdown/Build-options.md]: Oh hold on, looking at the implementation, you meant this to be a an array of free strings, not validated, hence the difference with #2390
I'm not sure what the use for this is, but sure, that looks alright then :)

File: docs/markdown/Build-options.md [hunk 2]
```diff
@@ -16,32 +16,37 @@ Here is a simple option file.
...
 
 ### Arrays
 
-Arrays allow one or more of the values in the `choices` parameter to be selected.
-If the `value` parameter is unset then the values of `choices` will be used as
-the default.
+Arrays represent an array of strings. By default the array can contain
+arbitrary strings. To limit the possible values that can used set the
+`choices` parameter. Meson will then only allow the value array to
+contain strings that are in the given list. The `value` parameter
```
Comment (dcbaker) [inline docs/markdown/Build-options.md]: It might be good to mention that the array can still be empty if `choices` are provided.

**Comment by jpakkane** [comment]
Merged manually.

**Pull request closed without merging**